plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/CurrentClassFQNProvider.java

```diff
@@ -54,8 +56,8 @@ public String getKey() {
     }
 
     @Override
-    public String getValue() {
-        return value;
+    public Promise<String> getValue() {
```

Comment:
The process of the configuring command line should be asynchronous, because some providers send request to the server for getting value (f.e. getting classpath).